### PR TITLE
Zappy 55 make ia

### PR DESCRIPTION
**Pull request by albe2**

### CODE REVIEW

File: client/ai/src/player.py [hunk 1]
```diff
@@ -178,21 +178,102 @@ def set_object(self, object : str) -> None:
...
+            self.right()
+            self.forward()
+        elif direction == 7:
+            self.right()
+            self.forward()
+        elif direction == 8:
+            self.right()
+            self.forward()
+            self.left()
+            self.forward()
```
Comment (Kapsulon) [inline client/ai/src/player.py]: make this into a list of lambdas

File: client/ai/src/player.py [hunk 2]
```diff
@@ -9,31 +9,60 @@ def __init__(self, team, machine, port):
...
-        for _ in range(pos.y):
-            self.forward()
-        if pos.x > 0:
-            self.right()
-        elif pos.x < 0:
-            self.left()
-        for _ in range(abs(pos.x)):
-            self.forward()
-        self.get_object(tile, searching_item)
+    level_requirements = {
```
Comment (Kapsulon) [inline client/ai/src/player.py]: constant variables should be ALL_CAPS

**Review by Kapsulon — CHANGES_REQUESTED** [review]
changes needed

File: client/ai/src/player.py (hunk 1 above)
Comment (Kapsulon) [reply client/ai/src/player.py]:
```py
MOVEMENT = [
    lambda: (self.forward()),
    lambda: (self.left(); self.forward(); self.right(); self.forward()),
    etc...
]

MOVEMENT[direction]()
```

Comment (albe2) [reply client/ai/src/player.py]: no problem

File: client/ai/src/player.py (hunk 2 above)
Comment (albe2) [reply client/ai/src/player.py]: no problem

File: client/ai/src/player.py [hunk 3]
```diff
@@ -9,31 +9,72 @@ def __init__(self, team, machine, port):
...
+            'deraumere': 2,
+            'sibur': 2,
+            'phiras': 2,
+            'mendiane': 2,
+            'thystame': 1
+        }
+    }
+
+
+    MOUVEMENTS_DIRECTION = [
```
Comment (Kapsulon) [inline client/ai/src/player.py]: french naming = cringe

**Review by Kapsulon — CHANGES_REQUESTED** [review]

**Review by Kapsulon — APPROVED** [review]
🫃🫃🫃

**Pull request merged**